### PR TITLE
fix: Attendee Form: Added Instagram and Linkedin

### LINKED ISSUE #6402: Attendee Form: Add Instagram and Linkedin
Instagram and Linkedin are missing as standard options on the atteendee form in the wizard. Please add them in the custom form section "Information to Collect".

Screenshot from 2021-01-17 17-58-29

### PR DESCRIPTION
**Pull request by afzal442**
Fixes #6402 

#### Changes proposed in this pull request:

- Added Instagram and Linkedin into Attendee Form
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/utils/validators.js [hunk 1]
```diff
@@ -70,6 +70,11 @@ export const validGithubProfileUrlPattern = new RegExp(
   + '?(?:www.)?github\\.com\\/([a-zA-Z0-9_]+)([-]?)([a-zA-Z0-9_]+)$'
 );
 
+export const validInstagramProfileUrlPattern = new RegExp(
+  '^(https?:\\/\\/)' // compulsory protocol
+  + '?(?:www.)?(instagram\\.com\\/([a-zA-Z0-9_.]+)$'
```
Comment (Creatoon) [inline app/utils/validators.js:75]:
```suggestion
  + '?(?:www.)?(instagram\\.com\\/)([a-zA-Z0-9_.]+)$'
```

**Review by Creatoon — CHANGES_REQUESTED** [review]

**Comment by afzal442** [comment]
SS After commmit
c1

**Comment by iamareebjamal** [comment]
Show SS of order form

**Comment by afzal442** [comment]
> Show SS of order form

But Sir. Issue is to add into attendee form.

File: app/components/forms/orders/order-form.js [hunk 2]
```diff
@@ -366,6 +366,56 @@ export default Component.extend(FormMixin, {
       ]
     };
 
+    const instagramValidation = {
```
Comment (iamareebjamal) [inline app/components/forms/orders/order-form.js:369]: Read the name of this file

**Comment by MeghalBisht** [comment]
> But Sir. Issue is to add into attendee form.

When linkedin/insta is included from the attendee form, it should appear on the order form. So issue by default includes it as well.
Hence, SS is asked for the order form.

**Comment by afzal442** [comment]
Thanks for your clarification@MeghalBisht . Will look into it.

**Comment by afzal442** [comment]
c2

**Review by MeghalBisht — COMMENTED** [review]
**Validation is not working**
snap

**"link" is missing**
op

**Validation works for facebook only**
Capture

File: app/utils/validators.js [hunk 3]
```diff
@@ -70,6 +70,11 @@ export const validGithubProfileUrlPattern = new RegExp(
   + '?(?:www.)?github\\.com\\/([a-zA-Z0-9_]+)([-]?)([a-zA-Z0-9_]+)$'
 );
 
+export const validInstagramProfileUrlPattern = new RegExp(
+  '^(https?:\\/\\/)' // compulsory protocol
+  + '?(?:www.)?(instagram\\.com\\/)([a-zA-Z0-9_.]+)$'
```
Comment (Creatoon) [inline app/utils/validators.js:75]:
```suggestion
  + '?(www.)?(?:instagram.com|instagr.am)\/[a-zA-Z0-9_.]+'
```

**Review by Creatoon — CHANGES_REQUESTED** [review]

**Comment by Creatoon** [comment]
Validatons will work now

Screenshot_2021-01-19 New Order - 546d488c-2c78-4136-95b9-ca1cb7dea5e7 Orders Open Event

File: app/utils/validators.js (hunk 3 above)
Comment (afzal442) [reply app/utils/validators.js:75]: Thanks for the suggestion. I assumed that.

**Comment by afzal442** [comment]
@MeghalBisht plz suggest I am not able to find proper regex for instagram that can work. Thanks.

**Comment by afzal442** [comment]
@Creatoon I think the issue in the server-side. Have a look in the [codebase.](https://github.com/fossasia/open-event-server/blob/5b84b00fc281c7eb48f82b8fcfd63738d87554d1/app/models/custom_form.py) See ATTENDEE_FORM, Instagram and LinkedIn are missing. Thanks.

**Comment by iamareebjamal** [comment]
Not really

**Comment by iamareebjamal** [comment]
> Validatons will work now
> 
> Screenshot_2021-01-19 New Order - 546d488c-2c78-4136-95b9-ca1cb7dea5e7 Orders Open Event

People have even shown fields being included in the form with some validations working and not working, and you are saying there is an issue in server?

It seems other people are more aware of the state of your PR than yourself!

**Comment by iamareebjamal** [comment]
@MeghalBisht @Creatoon Please check what is missing in the PR

**Comment by afzal442** [comment]
Nopes. Sir, u got me wrong. Have a look now.
c3
I have also inspected it's working but above one needs to be fixed. Thanks

**Comment by afzal442** [comment]
SS for latest commit
c2

**Comment by iamareebjamal** [comment]
> Nopes. Sir, u got me wrong. Have a look now.
> c3
> I have also inspected it's working but above one needs to be fixed. Thanks

The above image is unrelated to anything? Why should it be included or required by default? Is it written in the issue?

I simply asked to enable the fields and show me that they are included in the form with proper validation or not. I don't know what else you are solving in the issue

**Comment by afzal442** [comment]
Oops Sorry for that. Actually, I accidentally solved the PR by modifying the suggested code. Only Regex was the issue Thanks.

**Review by MeghalBisht — COMMENTED** [review]
Works for me!

**Comment by afzal442** [comment]
Thanks again. @MeghalBisht

**Comment by Creatoon** [comment]
Screenshot_2021-01-20 instagram profile - Regex Tester Debugger(3)


Your regex will also pass this url

https://instagr.am.com/username ,

it is not a valid url

File: app/utils/validators.js [hunk 4]
```diff
@@ -70,6 +70,11 @@ export const validGithubProfileUrlPattern = new RegExp(
   + '?(?:www.)?github\\.com\\/([a-zA-Z0-9_]+)([-]?)([a-zA-Z0-9_]+)$'
 );
 
+export const validInstagramProfileUrlPattern = new RegExp(
+  '^(https?:\\/\\/)' // compulsory protocol
+  + '?(?:www.)?(instagram|instagr.am)\\.com\\/([a-zA-Z0-9_.]+)$'
```
Comment (Creatoon) [inline app/utils/validators.js:75]: `  + '?(?:www.)?(instagram.com|instagr.am)\/([a-zA-Z0-9_.]+)$'`

**Pull request merged**